pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -378,6 +385,7 @@ private BrokerResponseNative handleRequest(long requestId, String query,
       if (_enableDistinctCountBitmapOverride) {
         handleDistinctCountBitmapOverride(serverPinotQuery);
       }
+      handleDistinctMultiValuedOverride(serverPinotQuery, getMultiValuedColumns(_tableCache, tableName));
```

Comment:
We can get schema first (move line 490 here) and pass schema into it if available
```
      Schema schema = _tableCache.getSchema(rawTableName);
      if (schema != null) {
        handleAggregationMVOverride(serverPinotQuery, schema);
      }
```